x-pack/legacy/plugins/maps/common/constants.js

```diff
@@ -50,7 +50,8 @@ export const ES_SPATIAL_RELATIONS = {
   INTERSECTS: 'INTERSECTS',
```

Comment:
I don't feel really strongly about this, but just not sure if we want to go _full GIS_ with a first pass at this. My inclination would be to start small.

I would just start with `INTERSECTS` as the only option. It's fine to show it in the dropdown with  a single option , to indicate that this maps to an ES-operation, but giving all these options makes it more convoluted.

- `disjoint` is just inverting an `intersects` filter. Inversion can be managed by the filter-bar. 
- the semantic difference between `contains` and `intersects` is minimal. There's no difference for point-filtering (any intersection will be containment), which I would expect would the 90% use-case (taking a shape and filtering points). Do users really (really) care?
- It's difficult to communicate the asymmetry of `within` and `contains`. I don't think `within` honestly makes a lot of sense with the current tooltip-driven UX. It would filter the other shapes that are surrounding the current shape. I think the intuition is more the other way around (ie. `contains`, create a filter for everything surrounded by this shape). Since contains isn't supported yet, I wouldn't add within either.

Reply:
wrt. within vs contains. I had them mixed up. within makes most sense.

Reply:
> disjoint is just inverting an intersects filter. Inversion can be managed by the filter-bar.

Inverting the filter-pill would require two fetches. The first fetch when the pill is created and the second fetch when the user  inverts the pill. I think it makes sense to allow users to set the relation from the UI.